fmt.go

```diff
@@ -90,6 +93,46 @@ func AvailableFormatters() map[string]string {
 	return fmts
 }
 
+// DelayedMessagesWriter is a structure used to print messages from a external library
+// This struct is useful when printing logs from the test themself
+// It will guarantee the orders of the log
+// For instance, when using this struct, the format will first print the step name and then all of the logs printed in this step
+type DelayedMessagesWriter struct {
```

Comment:
I do not see agree with this one. If you have read the [contribution guide](https://github.com/DATA-DOG/godog#contributions) in the end of the readme. you should have noticed that changes and additions to public interface are and will be treated carefully. I will not accept these unless they are really useful changes.

In this case, you are contributing a change which is not necessary. maybe it makes it easier for user to do it, but it adds redundancy to public godog interface and in time such useless features will only confuse people and inspire bad practices.

See the comment on main pull request, explaining how you should implement such behavior, because godog already gives you these options.